addons/l10n_in_hr_holidays/models/hr_leave.py

```diff
@@ -2,14 +2,28 @@
 
 from datetime import datetime, timedelta
 
-from odoo import models, fields
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
 
 
 class HrLeave(models.Model):
     _inherit = "hr.leave"
 
     l10n_in_contains_sandwich_leaves = fields.Boolean()
 
+    @api.constrains("holiday_status_id", "request_date_from", "request_date_to")
+    def _l10n_in_check_request_dates(self):
+        invalid_leaves = []
+        for leave in self:
+            if leave.holiday_status_id.l10n_in_is_limited_to_optional_days:
+                domain = [("date", "=", leave.request_date_from), ("date", "=", leave.request_date_to)]
+                if not self.env["l10n.in.hr.leave.optional.holiday"].search_count(domain, limit=1):
+                    invalid_leaves.append(f'    - {leave.display_name}')
+        if invalid_leaves:
+            raise ValidationError(
+                _("The following leave(s) are not on Optional Holidays:\n%s", "\n".join(invalid_leaves))
```

Comment:
```suggestion
                _("The following leave are not on Optional Holidays:\n%s", "\n".join(invalid_leaves))
```
The (s) like that is not really compatible with the translation :)